src/webui/api/searchcontroller.cpp

```diff
@@ -235,10 +239,16 @@ void SearchController::downloadTorrentAction()
     else
     {
         SearchDownloadHandler *downloadHandler = SearchPluginManager::instance()->downloadTorrent(pluginName, torrentUrl);
-        connect(downloadHandler, &SearchDownloadHandler::downloadFinished
-                , this, [this, downloadHandler](const QString &source)
+        connect(downloadHandler, &SearchDownloadHandler::downloadFinished, this
+            , [this, downloadHandler, pluginName, torrentUrl](const QString &source, const QString &errorMessage)
         {
+            if (!errorMessage.isEmpty())
+            {
+                LogMsg(tr("Error occurred when downloading torrent via search engine. Engine: \"%1\". URL: \"%2\". Error: \"%3\".")
```

Comment:
Maybe move error logging into search (download) handler itself?

Reply:
I suppose the handler should report the message to the caller rather than holding it to themselves.